include/oboe/Version.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OBOE_VERSIONINFO_H
+#define OBOE_VERSIONINFO_H
+
+namespace oboe {
+
+class Version {
+
+public:
+    // This is incremented when we make breaking API changes. Based loosely on https://semver.org/
+    static constexpr uint8_t MajorNumber = 0;
```

Comment:
I would argue that specifying the package version number is probably one (and maybe only) good application for the preprocessor. The reason is that with preprocessor, you can easily produce the version string at the build time (using the stringification feature). Whereas your current implementation requires the C library for that.

Potentially, this can be done using a constexpr function, but then you will need to write int to string conversion logic yourself, and this seems like an overkill (see https://stackoverflow.com/a/26824971/4477684).

Reply:
Yes I saw that whilst I was researching about whether we should switch to using `std::string`, it seems that you have to write some pretty advanced (read: difficult to read/maintain) C++ to produce compile-time string literals, something which is easy in C (with the down side of pointer/array duality).

Reply:
The static constants are now initialized using preprocessor directives. No need for any functions so I've removed the Version.cpp file.